src/main/java/io/cdap/plugin/gcp/gcs/sink/GCSBatchSink.java

```diff
@@ -383,6 +399,20 @@ public String getLocation() {
       return location;
     }
 
+    @Nullable
+    public String getContentType() {
+      if (!Strings.isNullOrEmpty(contentType)) {
+        if (contentType.equals(CONTENT_TYPE_OTHER)) {
+          if (Strings.isNullOrEmpty(customContentType)) {
+            return DEFAULT_CONTENT_TYPE;
+          }
+          return customContentType;
+        }
+        return contentType;
+      }
+      return DEFAULT_CONTENT_TYPE;
```

Comment:
Do we use default content type if the field is a macro?

Reply:
If the field is macro, the value that was specified by the user is being used.

Reply:
If this is a macro, during config time we will provide wrong value `DEFAULT_CONTENT_TYPE`. We should return null if there is a macro during config time.

Reply:
PR Updated.